File: app/src/main/java/dk/itu/moapd/mylocation/MapActivity.java
package dk.itu.moapd.mylocation;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

public class MapActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_map);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentById(R.id.fragment);

        if (fragment == null) {
            double longitude = getIntent().getDoubleExtra("longitude", 0);
            double latitude = getIntent().getDoubleExtra("latitude", 0);

            Bundle bundle = new Bundle();
            bundle.putDouble("longitude", longitude);
            bundle.putDouble("latitude", latitude);

            fragment = new MapFragment();
            fragment.setArguments(bundle);
            fragmentManager.beginTransaction()
                    .add(R.id.fragment, fragment)
                    .commit();
        }
    }

}

File: app/src/main/java/dk/itu/moapd/mylocation/MapActivity.kt
package dk.itu.moapd.mylocation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MapActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_map)

        var fragment = supportFragmentManager
                .findFragmentById(R.id.fragment)

        if (fragment == null) {
            val longitude = intent.getDoubleExtra("longitude", 0.0)
            val latitude = intent.getDoubleExtra("latitude", 0.0)

            val bundle = Bundle().apply {
                putDouble("longitude", longitude)
                putDouble("latitude", latitude)
            }

            fragment = MapFragment().apply {
                arguments = bundle
            }

            supportFragmentManager
                .beginTransaction()
                .add(R.id.fragment, fragment)
                .commit()
        }
    }

}
